lib/autoproj/package_managers/pip_manager.rb

```diff
@@ -25,39 +27,53 @@ def os_dependencies
             end
 
             def guess_pip_program
-                if Autobuild.programs['pip']
-                    return Autobuild.programs['pip']
+                if ws.config.has_value_for?('USE_PYTHON')
+                    unless ws.config.get('USE_PYTHON')
+                        raise ConfigError, "Your current package selection" \
+                          " requires the use of pip is required, but" \
+                          " the use of python has been denied, see" \
+                          " setting of USE_PYTHON in your workspace configuration." \
+                          " Either remove all packages depending on pip packages " \
+                          " from the workspace layout (manifest) or " \
+                          " call 'autoproj reconfigure' to change the setting."
+                    end
+                else
+                    Autoproj::Python.setup_python_configuration_options(ws: ws)
```

Comment:
You should error out here if the user chose not to use python...

Reply:
What exactly you mean? Replace "raising ConfigError"? Currently, a ConfigError is raised when a preference exists at all, i.e. USE_PYTHON is set?

Reply:
What I mean is that this should not happen:

```
arjones@781567b66194:~/test$ autoproj osdeps
  Do you want to activate python? [no] no
  installing/updating Python dependencies: mbed_cli, mbed_tools
  updated environment
Command finished successfully at 2021-05-30 18:04:30 -0300 (took 5 secs)
arjones@781567b66194:~/test$ 
```